server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -419,6 +419,49 @@ public void testDeleteOperations() throws Exception {
         }
     }
 
+    public void testDropPrimaryDuringReplication() throws Exception {
```

Comment:
I believe this test will rely on changes in https://github.com/opensearch-project/OpenSearch/pull/4225 or it will fail with resources left unclosed because it will force a cancellation that is not processed.

Reply:
Ah, got it. This test is only way to reproduce `no_such_file_exception` consistently. I did not observe file handle leaks related errors though (ran this test 20). 

Removing this.